diff -ru a/config.sub b/config.sub
--- a/config.sub	2001-09-09 01:46:40.000000000 +0000
+++ b/config.sub	2020-04-04 17:19:31.184929268 +0000
@@ -122,6 +122,10 @@
     os=-$maybe_os
     basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
+  *-android)
+    os=-linux-android
+    basic_machine=`echo $1 | grep -o '^[^-]*'`
+    ;;
   android-linux)
     os=-linux-android
     basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`-unknown
